src/components/api/whisper/whisperRecognizer.tsx

```diff
@@ -6,6 +6,17 @@ import { SIPOAudioBuffer } from './sipo-audio-buffer';
 import RecordRTC, {StereoAudioRecorder} from 'recordrtc';
 import WavDecoder from 'wav-decoder';
 
+// https://stackoverflow.com/questions/4554252/typed-arrays-in-gecko-2-float32array-concatenation-and-expansion
+function Float32Concat(first, second) {
+    const firstLength = first.length;
+    const result = new Float32Array(firstLength + second.length);
+
+    result.set(first);
+    result.set(second, firstLength);
+
+    return result;
+}
+
```

Comment:
This works but does a memory alloc every time a new chunk arrives (which also puts pressure on the garbage collector to run). Not a big deal but better code would reuse a buffer, and only allocate if the buffer was insufficient in size.